src/views/Home/components/LotteryCard.tsx

```diff
@@ -44,10 +47,28 @@ const FarmedStakingCard = () => {
   const lotteryHasDrawn = useGetLotteryHasDrawn()
   const [requesteClaim, setRequestedClaim] = useState(false)
   const TranslateString = useI18n()
+  const allowance = useLotteryAllowance()
+  const { onApprove } = useLotteryApprove()
+  const [onPresentApprove] = useModal(<PurchaseWarningModal />)
+  const [requestedApproval, setRequestedApproval] = useState(false)
   const { claimAmount } = useTotalClaim()
   const { onMultiClaim } = useMultiClaimLottery()
   const cakeBalance = useTokenBalance(getCakeAddress())
 
+  const handleApprove = useCallback(async () => {
```

Comment:
the whole code here has been copied from `TicketActions.tsx`. Please create a hook in order to avoid duplicate code.

Reply:
@RabbitDoge, I've refactored that code to a new hook with https://github.com/pancakeswap/pancake-frontend/pull/423/commits/f96f74d29e34eef3f82d80035ad2ac6b2f1e039c

As I'm unfamiliar with the best practices in this codebase, please let me know about naming conventions and where to put the hook itself. As it's used by two sibling views I've placed it in the `src/hooks`.